stablehlo/transforms/VhloToVersion.cpp

```diff
@@ -77,6 +78,69 @@ FailureOr<Version> validateTargetVersion(llvm::StringRef versionRef,
   return targetVersion;
 }
 
+template <typename AttrOrType>
+bool isDialect(AttrOrType at, llvm::StringRef dialectName) {
+  return (at.getDialect().getNamespace() == dialectName);
+}
+
+LogicalResult isLegalAttribute(Operation* op, NamedAttribute const& attr,
+                               Version targetVersion) {
+  // TODO: Remove once builtin types are forked.
+  if (isDialect(attr.getValue(), "builtin")) {
+    return success();
+  }
+
+  auto attrInterface = dyn_cast<VersionedAttrInterface>(attr.getValue());
+  if (attrInterface && isLegalVersionForTarget(attrInterface, targetVersion)) {
+    return success();
+  }
+
+  LLVM_DEBUG(llvm::dbgs() << "failed to legalize attribute " << attr.getName()
+                          << " to version " << targetVersion);
+  return failure();
+}
+
+LogicalResult isLegalType(Operation* op, Type const& type,
```

Comment:
I don't think it's idiomatic to use `const` or `&` with Type. The former is because https://mlir.llvm.org/docs/Rationale/UsageOfConst/, and the latter is because Type is lightweight enough to be passed by value.

Reply:
Hadn't seen that doc before, thanks! Fixed.